src/main/java/org/openjdk/jextract/impl/NameMangler.java

```diff
@@ -319,6 +319,11 @@ private static boolean isRestrictedTypeName(String name) {
     }
 
     private static boolean isJavaTypeName(String name) {
+        // rule out common case of lower-case starting identifier.
+        if (name.length() == 0 || Character.isLowerCase(name.charAt(0))) {
+            return false;
+        }
+
         // Java types that are used unqualified in the generated code
         return switch (name) {
             case "String", "Struct", "MethodHandle",
```

Comment:
should `String` be dropped from this list? Also, there seem to be other names that are out of sync, such as `NativeArena` and `Struct` (I wonder if we just want to repeat your trick for `java.lang.foreign`) and some are missing (e.g. `MethodHandle` but  not `VarHandle` ?)

Reply:
VarHandle and MethodHandle both are already included. Perhaps we can do same trick for java.lang.foreign.